folium/features.py

```diff
@@ -434,9 +434,20 @@ def style_data(self):
             self.data = {'type': 'FeatureCollection', 'features': [self.data]}
 
         for feature in self.data['features']:
-            feature.setdefault('properties', {}).setdefault('style', {}).update(self.style_function(feature))  # noqa
+
+            feature_style = self.style_function(feature)
+            new_style = {}
+            for item in feature_style:
+                if isinstance(feature_style[item], MacroElement):
+                    new_style[item] = "{{'" + feature_style[item].get_name() + "'}}"
+                else:
+                    new_style[item] = feature_style[item]
+
+            feature.setdefault('properties', {}).setdefault('style', {}).update(new_style)  # noqa
             feature.setdefault('properties', {}).setdefault('highlight', {}).update(self.highlight_function(feature))  # noqa
-        return json.dumps(self.data, sort_keys=True)
+
+        # TODO Avoid having to manually remove the quotes from from around the placeholders
+        return json.dumps(self.data, sort_keys=True).replace('"{{', "{{").replace('}}"', "}}")
```

Comment:
@Conengmo is more familiar with all the changes in the `GeoJson` class, so I'll defer to him to review this part :grimacing: